File: app/src/main/java/by/mrstark/mp3playerdemo/fragment/ArtistsFragment.java
package by.mrstark.mp3playerdemo.fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import by.mrstark.mp3playerdemo.R;
import by.mrstark.mp3playerdemo.adapter.ArtistsListAdapter;

/**
 * Created by mrstark on 27.1.16.
 */
public class ArtistsFragment extends AbstractTabFragment {

    public static ArtistsFragment getInstance(Context context) {
        ArtistsFragment fragment = new ArtistsFragment();
        fragment.setContext(context);
        fragment.setTitle(context.getString(R.string.tab_item_artists));
        return fragment;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    @Override
    public RecyclerView.Adapter getAdapter() {
        return new ArtistsListAdapter();
    }
}

File: app/src/main/java/by/mrstark/mp3playerdemo/fragment/ArtistsFragment.kt
package by.mrstark.mp3playerdemo.fragment

import android.content.Context
import android.os.Bundle
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import by.mrstark.mp3playerdemo.MainActivity

import by.mrstark.mp3playerdemo.R
import by.mrstark.mp3playerdemo.adapter.ArtistsListAdapter
import by.mrstark.mp3playerdemo.entity.Artist
import by.mrstark.mp3playerdemo.util.DataUtil

class ArtistsFragment : AbstractTabFragment() {
    interface OnArtistSelectedInterface {
        fun onArtistSelect(name: String)
    }

    override fun getAdapter(): RecyclerView.Adapter<RecyclerView.ViewHolder> {
        val listener = activity as OnArtistSelectedInterface
        return ArtistsListAdapter(listener, getArtists(arguments))
    }

    private fun getArtists(arguments: Bundle?): List<Artist>? {
        if (arguments != null) {
            val genreName = arguments.getString((activity as MainActivity).GENRE)
            return DataUtil.getInstace().artists.filter { it.getGenre().equals(genreName) }
        } else {
            return DataUtil.getInstace().artists
        }
    }

    fun getInstance(context: Context): ArtistsFragment {
        this.context = context
        this.title = context.getString(R.string.tab_item_artists)
        return this
    }
}
